src/utilities/extendSchema.js

```diff
@@ -171,204 +154,87 @@ export function extendSchema(
     return schema;
   }
 
-  const definitionBuilder = new ASTDefinitionBuilder(
+  const astBuilder = new ASTDefinitionBuilder(
     typeDefinitionMap,
     options,
-    (typeName, node) => {
-      const existingType = schema.getType(typeName);
-      if (existingType) {
-        return extendType(existingType);
-      }
+    typeRef => {
+      invariant(schemaTransformer);
+      const typeName = typeRef.name.value;
+      const type = schemaTransformer.transformType(typeName);
```

Comment:
This API is a bit confusing - this isn't a transform function, it's something that takes a name and return a type, so I'd expect this to be more like the `getType()` API